### PR TITLE
deploy with app collection

### PR DESCRIPTION
**Pull request by TheoBrigitte**
Towards: https://github.com/giantswarm/giantswarm/issues/6847

This PR configures CircleCI to deploy tagged version of aws-operator.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -19,7 +19,7 @@ e2eTest: &e2eTest
         when: always
 
 orbs:
-  architect: giantswarm/architect@0.2.0
+  architect: giantswarm/architect@dev:5c92e1f
```
Comment (TheoBrigitte) [inline .circleci/config.yml]: Note: update to latest version before merging

File: .circleci/config.yml [hunk 2]
```diff
@@ -252,6 +253,17 @@ workflows:
           filters:
             tags:
               only: /^v.*/
+      - architect/push-to-app-collection:
+          name: push-aws-operator-to-app-collection
```
Comment (kopiczko) [inline .circleci/config.yml]:
```suggestion
          name: push-to-aws-app-collection
```

File: .circleci/config.yml [hunk 3]
```diff
@@ -252,6 +253,17 @@ workflows:
           filters:
             tags:
               only: /^v.*/
+      - architect/push-to-app-collection:
+          name: push-to-aws-app-collection
+          app_name: "aws-operator"
+          app_collection_repo: "aws-app-collection"
```
Comment (kopiczko) [inline .circleci/config.yml]: Don't we need app_catalog somewhere here?

Comment (TheoBrigitte) [reply .circleci/config.yml]: default value is `control-plane-catalog`

**Pull request merged**